src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/routinghelper/RoutingHelperPanel.java

```diff
@@ -0,0 +1,97 @@
+package org.openstreetmap.josm.plugins.pt_assistant.actions.routinghelper;
+
+import java.awt.BorderLayout;
+import java.awt.Color;
+import java.awt.Component;
+import java.awt.Cursor;
+import java.awt.FlowLayout;
+import java.util.Optional;
+
+import javax.swing.JButton;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.border.EmptyBorder;
+
+import com.drew.lang.annotations.Nullable;
+import org.openstreetmap.josm.data.osm.Way;
+import org.openstreetmap.josm.gui.MainApplication;
+import org.openstreetmap.josm.gui.MapFrame;
+import org.openstreetmap.josm.tools.I18n;
+import org.openstreetmap.josm.tools.ImageProvider;
+
+/**
+ * The top panel that is added via {@link MapFrame#addTopPanel(Component)}.
+ * This class should just handle the display and input. The state of the routing helper should be handled in {@link RoutingHelperAction}.
+ */
+public class RoutingHelperPanel extends JPanel {
```

Comment:
This panel handles the display of the top panel (similar to the snackbar we had a while ago). When something in the routing helper changes it is updated through e.g. `onCurrentWayChange()` below.

In turn this panel notifies the RoutingHelperAction when a button is pressed.